### PR TITLE
Add `text/markdown` mimeType lookup by extension

### PR DESCRIPTION
**Pull request by literallylara**
Fixes https://github.com/dart-lang/mime/issues/1

RFC: https://www.rfc-editor.org/rfc/rfc7763.html

### CODE REVIEW

**Comment by literallylara** [comment]
Turns out `text/markdown` is [official](https://www.rfc-editor.org/rfc/rfc7763.html) since 2016, so the `x-` can be dropped. Not sure if this can be changed while merging?

File: lib/src/default_extension_map.dart [hunk 1]
```diff
@@ -434,6 +434,7 @@ const Map<String, String> defaultExtensionMap = <String, String>{
   'mc1': 'application/vnd.medcalcdata',
   'mcd': 'application/vnd.mcd',
   'mcurl': 'text/vnd.curl.mcurl',
+  'md': 'text/markdown',
```
Comment (kevmoo) [inline lib/src/default_extension_map.dart:437]:
```suggestion
  // https://www.rfc-editor.org/rfc/rfc7763
  'md': 'text/markdown',
```

Going forward, I think it'd be good to include these in all additions. Keeps everyone honest!

**Review by kevmoo — APPROVED** [review]
Tests? Changelog change? With version bump?

#swoon

**Comment by kevmoo** [comment]
@devoncarew @natebosch – thoughts?

File: lib/src/default_extension_map.dart [hunk 2]
```diff
@@ -434,6 +434,8 @@ const Map<String, String> defaultExtensionMap = <String, String>{
   'mc1': 'application/vnd.medcalcdata',
   'mcd': 'application/vnd.mcd',
   'mcurl': 'text/vnd.curl.mcurl',
+  // https://www.rfc-editor.org/rfc/rfc7763
```
Comment (natebosch) [inline lib/src/default_extension_map.dart:437]: The RFS also mentions `.markdown` as an extension. Do we think that's unpopular enough now to not be worth supporting?

Comment (literallylara) [reply lib/src/default_extension_map.dart:437]: Good catch! In 2014, one of the authors of Markdown actually [endorsed](https://web.archive.org/web/20200712120733/https://daringfireball.net/linked/2014/01/08/markdown-extension) `.markdown` as the only extension they would use. Either way, I think it makes sense to add for the sake of completeness. I'll prepare a commit.

**Review by natebosch — APPROVED** [review]

**Pull request merged**